performance_test.js

```diff
@@ -0,0 +1,40 @@
+const path = require('path');
+const xmldom = require('xmldom');
+const fs = require('fs');
+const { performance } = require('perf_hooks');
+const { VASTParser } = require('./dist/vast-client-node.min.js');
+
+const getXmlFromFilename = filename => {
+  const url = path
+    .resolve(path.join('spec', 'samples', filename))
+    .replace(/\\/g, '/');
+
+  const stringifiedXml = fs.readFileSync(url).toString();
+  const parser = new xmldom.DOMParser();
+  return parser.parseFromString(stringifiedXml, 'text/xml');
+};
+
+const getParsingTime = (VastParser, xml) => {
+  const timeBefore = performance.now();
+  return VastParser.parseVAST(xml).then(() => {
+    const timeAfter = performance.now();
+    return timeAfter - timeBefore;
+  });
+};
+
+const getAverage = list => {
+  return list.reduce((res, value) => (res += value), 0) / list.length;
+};
+
+const inlineXml = getXmlFromFilename('inline-linear.xml');
+const VastParser = new VASTParser();
+const numberOfRuns = 500;
+const arr = new Array(numberOfRuns).fill(0);
+
+const promises = arr.map(() => getParsingTime(VastParser, inlineXml));
+
+Promise.all(promises).then(results => {
+  const average = getAverage(results);
```

Comment:
Problem: The higher the `numberOfRuns` the higher the average becomes. I have changed from `Date.now()` to `performance.now()` and it didn't solve the issue. Any inputs here would be helpful :)

Reply:
Solution: Each loop runs the test in a separate process so they don't share resources with other tests and become slower with time. `Promise.all` is also not reliable because it runs all the promises, and only after it has executed all the promises, will it begin to resolve them (so the timing isn't right). Now we wait for each run to be resolved before going to the next one. Downside is that the tests are a bit slow now.